src/behaviours/pod-logs/download-logs.test.tsx

```diff
@@ -74,58 +89,76 @@ describe("download logs options in pod logs dock tab", () => {
 
       openSaveFileDialogMock = jest.fn();
       windowDi.override(openSaveFileDialogInjectable, () => openSaveFileDialogMock);
+
+      showErrorNotificationMock = jest.fn();
+      windowDi.override(showErrorNotificationInjectable, () => showErrorNotificationMock);
     });
   });
 
-  describe("when opening pod logs", () => {
+  describe("opening pod logs", () => {
     beforeEach(async () => {
       rendered = await builder.render();
-
-      const windowDi = builder.applicationWindow.only.di;
-      const pod = dockerPod;
-      const createLogsTab = windowDi.inject(createPodLogsTabInjectable);
-      const container = {
-        name: "docker-exporter",
-        image: "docker.io/prom/node-exporter:v1.0.0-rc.0",
-        imagePullPolicy: "pull",
-      };
-
+      windowDi = builder.applicationWindow.only.di;
+  
       const dockStore = windowDi.inject(dockStoreInjectable);
 
       dockStore.closeTab("terminal");
+    });
 
-      createLogsTab({
-        selectedPod: pod,
-        selectedContainer: container,
+    describe("when logs not available", () => {
+      beforeEach(async () => {
+        const createLogsTab = windowDi.inject(createPodLogsTabInjectable);
+    
+        getLogsMock.mockReturnValue([]);
+        getSplittedLogsMock.mockReturnValue([]);
+  
+        createLogsTab({
+          selectedPod: pod,
+          selectedContainer: container,
+        });
       });
```

Comment:
I think these improvements are great! I think that if we used `AsyncFnMock` then we would be able to get more fine grained specification of the behaviour.

For example:

```ts
describe("when a pod logs tab is created", () => {
  beforeEach(() => {
    const createLogsTab = windowDi.inject(createPodLogsTabInjectable);

    createLogsTab({
      selectedPod: pod,
      selectedContainer: container,
    });
  });

  it("should request pod logs", () => {
    expect(callForLogsMock).toBeCalledWith(...);
  });

  it.todo("should open the dock tab");

  it.todo("should show a spinner");

  describe("when the pod logs request resolves to empty", () => {
    beforeEach(async () => {
      await callForLogsMock.resolve("");
    });
    
    it.todo("the download pod logs dropdown should be disabled");

    ...
  });

  describe("when the pod logs request resolves to some lines", () => {
    beforeEach(async () => {
      await callForLogsMock.resolve("a log line");
    });

    it.todo("the download pod logs dropdown should be enabled");

    ...
  });
});
```

Reply:
I don't see why using AsyncFnMock would be more useful in our case. For now mocking resulted value done using following form:
```
beforeEach(async () => {
  callForLogsMock.mockResolvedValue("all-logs");
});
```

this is still quite readable and not very different from this form

```
beforeEach(async () => {
  await callForLogsMock.resolve("all-logs");
});
```

Reply:
The difference is the ability to specify in the tests the expected behaviour while the request has not yet resolved.

That is not possible with the `callForLogsMock.mockResolvedValue` version as it will throw if such a mock is not in place (or just immediately return with `undefined`).